### PR TITLE
Feature/financial categories tabs

### PR DESCRIPTION
**Pull request by nesharim**
Will add the redux part a little later. As of now build the page so that it has an almost close match to the mock up. Also after approval will get this merged once Nav bar PR has been merged.

edit

main

row-deleted

### CODE REVIEW

File: client/src/App.js [hunk 1]
```diff
@@ -1,12 +1,16 @@
...
+import FinancialCategoryPage from "views/financial-categories/pages/main";
+
 export default function App() {
   return (
     <div className="App">
-      <CssBaseline />
+      {/* <CssBaseline />
       <h1>Welcome to Project Paycheck</h1>
-      <PaycheckCard/>
+      <PaycheckCard/> */}
```
Comment (ghost) [inline client/src/App.js]: Typically I would tell you to not include commented-out code in your PRs but seeing how the routes are not done, it's fine

File: client/public/index.html [hunk 2]
```diff
@@ -12,6 +12,7 @@
       rel="stylesheet"
       href="https://fonts.googleapis.com/css?family=Roboto:300,400,500,700&display=swap"
     />
+    <link rel="stylesheet" href="https://fonts.googleapis.com/icon?family=Material+Icons" />
```
Comment (ghost) [inline client/public/index.html]: Please remove. We already have material icons set up as a dependency, check the package.json file
"@material-ui/icons": "^4.2.1"

File: client/src/views/financial-categories/forms/category-form.jsx [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+  return (
+    <Card className={classes.card}>
+      <MaterialTable
+        title={title}
+        columns={layout}
+        data={state.data}
+        editable={{
+          onRowAdd: newData =>
+            new Promise(resolve => {
+              setTimeout(() => {
```
Comment (ghost) [inline client/src/views/financial-categories/forms/category-form.jsx]: The DOCs say to return a promise so I understand why you did this, but this is kinda ugly and hard to read. Please change to use Async/Await. See if this works

```javascript
onRowUpdate: async (newData, oldData) => {
  const data = [...state.data];
  data[data.indexOf(oldData)] = newData;
  setState({ ...state, data });
}
```

File: client/src/views/financial-categories/forms/category-form.jsx [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+            }),
+          onRowUpdate: (newData, oldData) =>
+            new Promise(resolve => {
+              setTimeout(() => {
+                resolve();
+                const data = [...state.data];
+                data[data.indexOf(oldData)] = newData;
+                setState({ ...state, data });
+              }, 600);
+            }),
```
Comment (ghost) [inline client/src/views/financial-categories/forms/category-form.jsx]: could you move these functions out of the return block and make them callback functions that you pass to it? It'll make it easier to read.

File: client/src/views/financial-categories/forms/category-form.jsx [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+FinancialCategoryForm.prototypes = {
+  title: PropTypes.string.isRequired,
+  layout: PropTypes.arrayOf(
+    PropTypes.shape({
+      title: PropTypes.string.isRequired,
+      field: PropTypes.string.isRequired,
+      type: PropTypes.string
+    })
+  ).isRequired
+};
```
Comment (ghost) [inline client/src/views/financial-categories/forms/category-form.jsx:83]: Thank you for using propTypes :D

**Review by ghost — CHANGES_REQUESTED** [review]

**Comment by nesharim** [comment]
@Dichotomyy ready for review

**Pull request merged**